esp-hal/src/spi/master.rs

```diff
@@ -2737,68 +2739,162 @@ impl Driver {
     }
 
     /// Enable or disable listening for the given interrupts.
-    #[cfg(gdma)]
     fn enable_listen(&self, interrupts: EnumSet<SpiInterrupt>, enable: bool) {
         let reg_block = self.register_block();
 
-        reg_block.dma_int_ena().modify(|_, w| {
-            for interrupt in interrupts {
-                match interrupt {
-                    SpiInterrupt::TransDone => w.trans_done().bit(enable),
-                    SpiInterrupt::DmaInfifoFullErr => w.dma_infifo_full_err().bit(enable),
-                    SpiInterrupt::DmaOutfifoEmptyErr => w.dma_outfifo_empty_err().bit(enable),
-                    SpiInterrupt::DmaSegTransDone => w.dma_seg_trans_done().bit(enable),
-                    SpiInterrupt::SegMagicErr => w.seg_magic_err().bit(enable),
-                    SpiInterrupt::RxAfifoWfullErr => w.mst_rx_afifo_wfull_err().bit(enable),
-                    SpiInterrupt::TxAfifoRemptyErr => w.mst_tx_afifo_rempty_err().bit(enable),
-                    SpiInterrupt::App2 => w.app2().bit(enable),
-                    SpiInterrupt::App1 => w.app1().bit(enable),
-                };
+        cfg_if::cfg_if! {
+            if #[cfg(gdma)] {
```

Comment:
It's an extreme nitpick, but this shouldn't be `gdma`, otherwise we'll have to duplicate this for the P4. I'd recommend `if esp32 {} else if s2 {} else { all others }`